script.js

```diff
@@ -1,4 +1,213 @@
+    user1Numbers = user1Numbers.sort(function (a, b) {
+      return a - b;
+    });
+    var counter = 0;
+    while (counter < user1Numbers.length) {
+      user1Number +=
+        user1Numbers[counter] * 10 ** (NumberofDices - 1 - counter);
+      counter += 1;
+    }
+  }
+  user1SCORE = user1SCORE + user1Number;
+  console.log(user1SCORE);
+
+  return `User 1, your number is ${user1Number}`;
+};
+
+///roll the dice for user2's two numbers
+var part2 = function () {
+  var roll = 0;
+  while (roll < NumberofDices) {
+    var user2Num = rollDice();
+    user2Numbers.push(user2Num);
+    roll += 1;
+  }
+
+  if (gamemode == "Normal") {
+    user2Numbers = user2Numbers.sort(function (a, b) {
+      return b - a;
+    });
+
+    var counter = 0;
+    while (counter < user2Numbers.length) {
+      user2Number +=
+        user2Numbers[counter] * 10 ** (NumberofDices - 1 - counter);
+      counter += 1;
+    }
+  } else if (gamemode == "Reverse") {
+    user2Numbers = user2Numbers.sort(function (a, b) {
+      return a - b;
+    });
+    var counter = 0;
+    while (counter < user2Numbers.length) {
+      user2Number +=
+        user2Numbers[counter] * 10 ** (NumberofDices - 1 - counter);
+      counter += 1;
+    }
+  }
+  user2SCORE = user2SCORE + user2Number;
+  console.log(user2SCORE);
+
+  return `User 2, your number is ${user2Number}`;
+};
+
+///get the leaderboard for the reverse mode.
+var reverseleaderboard = function (user1SCORE, user2SCORE) {
+  var message = "";
+  if (user1SCORE <= user2SCORE) {
+    message = `1. User1's Score: ${user1SCORE} <br>2. User2's Score: ${user2SCORE}`;
+  } else if (user2SCORE < user1SCORE) {
+    message = `1. User2's Score: ${user2SCORE} <br>2. User1's Score: ${user1SCORE}`;
+  }
+  return (
+    message +
+    '<br><br>Click the "submit" button to replay. <br>To change Game Modes, please refresh page.'
+  );
+};
+
+///get the outcome for the reverse mode (e.g. win or lose)
+var reverseOutcome = function (user1Number, user2Number) {
+  var message = `Player 1's number is ${user1Number} while Player 2's number is ${user2Number}. <br>`;
+
+  if (user1Number < user2Number) {
+    message += `User 1 won! Try again later User 2. <br>`;
+  } else if (user2Number > user1Number) {
+    message += `User 2 won! Try again later User 1. <br>`;
+  } else if (user1Number == user2Number) {
+    message += `It's a tie! <br>Better luck next time. <br>`;
+  }
+  message += `<br>`;
+  return message;
+};
+
+///get the leaderboard for the normal mode.
+var normalleaderboard = function (user1SCORE, user2SCORE) {
+  var message = "";
+  if (user1SCORE >= user2SCORE) {
+    message = `1. User1's Score: ${user1SCORE} <br>2. User2's Score: ${user2SCORE}`;
+  } else if (user2SCORE > user1SCORE) {
+    message = `1. User2's Score: ${user2SCORE} <br>2. User1's Score: ${user1SCORE}`;
+  }
+  return (
+    message +
+    '<br><br>Click the "submit" button to replay. <br>To change Game Modes, please refresh page.'
+  );
+};
```

Comment:
same issue, this logic is the same as var reverseOutcome = function (user1Number, user2Number), reuse it instead of typing the same logic